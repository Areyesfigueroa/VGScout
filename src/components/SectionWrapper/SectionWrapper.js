import React from "react";
import classes from "./SectionWrapper.module.css";

const sectionWrapper = (props) => (
    <section className={classes.SectionWrapper}>
        {props.children}
    </section>
);

export default sectionWrapper;