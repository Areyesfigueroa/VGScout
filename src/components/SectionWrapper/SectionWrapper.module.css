.SectionWrapper {
    margin-bottom: 10px;
    box-sizing: border-box;
}