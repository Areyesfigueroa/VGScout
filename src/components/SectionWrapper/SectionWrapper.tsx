import React from "react";
import classes from "./SectionWrapper.module.css";

const sectionWrapper = (props) => (
    <section data-testid={'sectionWrapper'} className={classes.SectionWrapper}>
        {props.children}
    </section>
);

export default sectionWrapper;