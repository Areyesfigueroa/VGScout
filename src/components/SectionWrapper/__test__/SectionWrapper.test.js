import React from "react";
import ReactDOM from "react-dom";

//React's UI testing library
import { render,cleanup, getByTestId } from '@testing-library/react';
import {toBeInTheDocument, toHaveStyle} from '@testing-library/jest-dom';

//Snapshots
import renderer from 'react-test-renderer';

//testing components
import SectionWrapper from '../SectionWrapper';

expect.extend({toBeInTheDocument, toHaveStyle});
afterEach(cleanup);

it("render without crashing", () => {
    const { getByTestId } = render(<SectionWrapper></SectionWrapper>);
    expect(getByTestId('sectionWrapper').classList.contains("SectionWrapper")).toBeTruthy();
});

it("render children props without crashing", () => {
    render(<SectionWrapper>Children Props</SectionWrapper>);
});

//Snapshot Testing
it("Matches snapshot 1", () => {
    const tree = renderer.create(<SectionWrapper></SectionWrapper>).toJSON();
    expect(tree).toMatchSnapshot();
});

it("Matches snapshot 2", () => {
    const tree = renderer.create(<SectionWrapper><div>Anything</div></SectionWrapper>).toJSON();
    expect(tree).toMatchSnapshot();
});