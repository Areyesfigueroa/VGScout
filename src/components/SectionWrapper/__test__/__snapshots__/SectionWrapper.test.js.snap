// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1 1`] = `
<section
  className="SectionWrapper"
  data-testid="sectionWrapper"
/>
`;

exports[`Matches snapshot 2 1`] = `
<section
  className="SectionWrapper"
  data-testid="sectionWrapper"
>
  <div>
    Anything
  </div>
</section>
`;
